### PR TITLE
[22.01] Drop pkg_resources

### PR DESCRIPTION
**Pull request by mvdbeek**
and other legacy stuff.

This speeds up importing python modules considerably and together with another PR would allow us to import Galaxy's config in other packages. 

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/exceptions/error_codes.py [hunk 1]
```diff
@@ -45,7 +43,7 @@ def _from_dict(entry):
     return (name, ErrorCode(code, message))
 
 
-error_codes_json = unicodify(resource_string(__name__, 'error_codes.json'))
+error_codes_json = resource_string(__name__.rsplit('.', 1)[0], 'error_codes.json')
```
Comment (nsoranzo) [inline lib/galaxy/exceptions/error_codes.py]:
```suggestion
error_codes_json = resource_string(__package__, "error_codes.json")
```

File: lib/galaxy/managers/licenses.py [hunk 2]
```diff
@@ -85,7 +85,7 @@ class LicenseMetadataModel(BaseModel):
     "MPL-2.0",
     "PDDL-1.0",
 ]
-SPDX_LICENSES_STRING = resource_string(__name__, 'licenses.json').decode("UTF-8")
+SPDX_LICENSES_STRING = resource_string(__name__.rsplit('.', 1)[0], 'licenses.json')
```
Comment (nsoranzo) [inline lib/galaxy/managers/licenses.py]:
```suggestion
SPDX_LICENSES_STRING = resource_string(__package__, "licenses.json")
```

File: lib/galaxy/managers/markdown_util.py [hunk 3]
```diff
@@ -596,7 +596,7 @@ def to_pdf_raw(basic_markdown: str, css_paths: Optional[List[str]] = None) -> by
         output_file.write(as_html)
         output_file.close()
         html = weasyprint.HTML(filename=index)
-        stylesheets = [weasyprint.CSS(string=pkg_resources.resource_string(__name__, 'markdown_export_base.css'))]
+        stylesheets = [weasyprint.CSS(string=resource_string(resource_string, 'markdown_export_base.css'))]
```
Comment (nsoranzo) [inline lib/galaxy/managers/markdown_util.py]:
```suggestion
        stylesheets = [weasyprint.CSS(string=resource_string(__package__, "markdown_export_base.css"))]
```

File: lib/galaxy/selenium/data.py [hunk 4]
```diff
@@ -1,10 +1,10 @@
 import yaml
-from pkg_resources import resource_string
 
+from galaxy.util.resources import resource_string
 from .components import Component
 
 
 def load_root_component() -> Component:
-    new_data_yaml = resource_string(__name__, 'navigation.yml').decode("UTF-8")
+    new_data_yaml = resource_string(__name__.rsplit('.', 1)[0], 'navigation.yml')
```
Comment (nsoranzo) [inline lib/galaxy/selenium/data.py]:
```suggestion
    new_data_yaml = resource_string(__package__, "navigation.yml")
```

File: lib/galaxy/tools/__init__.py [hunk 5]
```diff
@@ -210,7 +210,7 @@
     "winSplitter": packaging.version.parse("1.0.1"),
 }
 
-BIOTOOLS_MAPPING_CONTENT = resource_string(__name__, 'biotools_mappings.tsv').decode("UTF-8")
+BIOTOOLS_MAPPING_CONTENT = resource_string(__name__, 'biotools_mappings.tsv')
```
Comment (nsoranzo) [inline lib/galaxy/tools/__init__.py]:
```suggestion
BIOTOOLS_MAPPING_CONTENT = resource_string(__package__, "biotools_mappings.tsv")
```

File: lib/galaxy/util/rules_dsl.py [hunk 6]
```diff
@@ -4,11 +4,12 @@
...
 
 import yaml
-from pkg_resources import resource_stream
+
+from galaxy.util.resources import resource_string
 
 
 def get_rules_specification():
-    return yaml.safe_load(resource_stream(__name__, 'rules_dsl_spec.yml'))
+    return yaml.safe_load(resource_string(__name__.rsplit('.', 1)[0], 'rules_dsl_spec.yml'))
```
Comment (nsoranzo) [inline lib/galaxy/util/rules_dsl.py]:
```suggestion
    return yaml.safe_load(resource_string(__package__, "rules_dsl_spec.yml"))
```

File: lib/galaxy_test/base/populators.py [hunk 7]
```diff
@@ -83,19 +82,19 @@
...
+from galaxy.util.resources import resource_string
 from . import api_asserts
 from .api import ApiTestInteractor
 
 
 CWL_TOOL_DIRECTORY = os.path.join(galaxy_root_path, "test", "functional", "tools", "cwl_tools")
 
 # Simple workflow that takes an input and call cat wrapper on it.
-workflow_str = unicodify(resource_string(__name__, "data/test_workflow_1.ga"))
+workflow_str = resource_string(__name__.rsplit('.', 1)[0], "data/test_workflow_1.ga")
```
Comment (nsoranzo) [inline lib/galaxy_test/base/populators.py]:
```suggestion
workflow_str = resource_string(__package__, "data/test_workflow_1.ga")
```

File: lib/galaxy_test/base/populators.py [hunk 8]
```diff
@@ -83,19 +82,19 @@
...
 # Simple workflow that takes an input and filters with random lines twice in a
 # row - first grabbing 8 lines at random and then 6.
-workflow_random_x2_str = unicodify(resource_string(__name__, "data/test_workflow_2.ga"))
+workflow_random_x2_str = resource_string(__name__.rsplit('.', 1)[0], "data/test_workflow_2.ga")
```
Comment (nsoranzo) [inline lib/galaxy_test/base/populators.py]:
```suggestion
workflow_random_x2_str = resource_string(__package__, "data/test_workflow_2.ga")
```

File: lib/galaxy_test/base/populators.py [hunk 9]
```diff
@@ -1124,7 +1123,7 @@ def load_random_x2_workflow(self, name: str) -> dict:
     def load_workflow_from_resource(self, name: str, filename: Optional[str] = None) -> dict:
         if filename is None:
             filename = f"data/{name}.ga"
-        content = unicodify(resource_string(__name__, filename))
+        content = resource_string(__name__.rsplit('.', 1)[0], filename)
```
Comment (nsoranzo) [inline lib/galaxy_test/base/populators.py]:
```suggestion
        content = resource_string(__package__, filename)
```

File: lib/galaxy/webapps/__init__.py [hunk 10]
```diff
@@ -1,3 +1,4 @@
 """Galaxy webapps root package -- this is a namespace package."""
 
-__import__("pkg_resources").declare_namespace(__name__)
+from pkgutil import extend_path
+__path__ = extend_path(__path__, __name__)  # type: ignore[has-type]
```
Comment (nsoranzo) [inline lib/galaxy/webapps/__init__.py:5]:
```suggestion

__path__ = extend_path(__path__, __name__)  # type: ignore[has-type]
```

**Comment by mvdbeek** [comment]
Let's give it a try

**Review by jmchilton — APPROVED** [review]

**Pull request merged**